### DISCUSSION TITLE
optional enforcement for lint/complexity/useLiteralKeys

### DISCUSSION THREAD

**Discussion by yckbilly1929** [biomejs/biome#327] [Ideas]
After I switched from rome to biome, there are some changes to default behavior, like this one:
`The computed expression can be simplified without the use of a string literal. biome lint/complexity/useLiteralKeys`

Would like to know if it is possible to provide a config option to separate the checking of literal keys for dottable type and not-quite-dottable type (e.g. `{ [key: string]: any }` ) ? Thanks.

**Comment by Conaclos** [comment]
Do you mean that the behavior should depend on how is typed a property? Unfortunately we have no access to type information. I mainly accept with your point. I was an early adopter of TypeScript's [noPropertyAccessFromIndexSignature](https://www.typescriptlang.org/tsconfig/#noPropertyAccessFromIndexSignature). I finally turned off this config because it conflicts with Biome's `useLiteralKeys`.

After some thoughts, I concluded that it is not a good idea of using index signatures. In most of the cases, a Map should be used. You can also completely disable the rule in your config file.

**Reply by yckbilly1929** [reply]
Thanks for your reply. I agree that using index signatures is not a good idea, will try to use a Map instead. 👍
